test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9576,6 +9637,37 @@ def create_vnodes(self, name=None, attrib=None, num=1, mom=None,
             self.logger.error("name, attributes, and mom object are required")
             return False
 
+        if mom.is_cpuset_mom():
+            self.check_mem_request(attrib)
+            if len(attrib) == 0:
+                return True
+            nodes = self.status(HOST, id=mom.shortname)
+            del nodes[0]  # don't set any attribute on natural node
+            if len(nodes) < num:
+                msg = 'cpuset mom does not have required number of nodes'
+                self.skipTest(msg)
+            elif len(nodes) == num:
+                if attrib:
+                    for vnode in nodes:
+                        self.check_ncpus_request(attrib, vnode)
+                        if len(attrib) == 0:
+                            return True
+                        self.set_node_attrib(vnode, attrib)
+                return True
+            else:
+                i = 0
+                for vnode in nodes:
+                    if i < num:
+                        i += 1
+                        self.check_ncpus_request(attrib, vnode)
+                        if len(attrib) == 0:
+                            return True
```

Comment:
I'm trying to understand this block of code. I assume 'attrib' is something that needs to be set on all the vnodes that satisfy the # of vnodes requested. So for the extra vnodes, we offline. But here, I see a 'return True' if ncpus request is already satisfied by a vnode < the # of vnodes requested. Am I getting this confused? It doesn't look like the extra vnodes would get offlined in this case...

Reply:
Thanks for catching it. I added this code late, didnt check the implications. Will modify the condition to set the attrib if len(attrib) is not 0.

Reply:
@visheshh : has this been addressed?

Reply:
@bayucan : Yes Al, This has been addressed in create_vnodes. In Manager function however we should return true as we wont be having anything else to do if there are no attributes to set.